nautobot/docs/development/apps/api/platform-features/populating-extensibility-features.md

```diff
@@ -1,10 +1,14 @@
-# Populating Extensibility Features
-
-+++ 1.2.0
+# Populating Extensibility Feature And Base Data
 
 In many cases, an app may wish to make use of Nautobot's various extensibility features, such as [custom fields](../../../../user-guide/platform-functionality/customfield.md) or [relationships](../../../../user-guide/platform-functionality/relationship.md). It can be useful for an app to automatically create a custom field definition or relationship definition as a consequence of being installed and activated, so that everyday usage of the app can rely upon these definitions to be present.
 
-To make this possible, Nautobot provides a custom [signal](https://docs.djangoproject.com/en/stable/topics/signals/), `nautobot_database_ready`, that apps can register to listen for. This signal is triggered when `nautobot-server migrate` or `nautobot-server post_upgrade` is run after installing an app, and provides an opportunity for the app to make any desired additions to the database at this time.
+Furthermore, sometimes apps might want to create a baseline of available data. This could, for example, be a baseline of circuit providers for a [Single Source of Truth](https://docs.nautobot.com/projects/ssot/en/latest/) integration that synchronizes circuit data or creating new default [statuses](https://docs.nautobot.com/projects/core/en/stable/user-guide/platform-functionality/status/) and/or [roles](https://docs.nautobot.com/projects/core/en/stable/user-guide/platform-functionality/role/), or adding to the allowed content types for existing statuses and roles.
+
+## Using A Signal Handler
+
++++ 1.2.0
+
+In general, it is recommended to use custom [signal](https://docs.djangoproject.com/en/stable/topics/signals/), `nautobot_database_ready` that apps can register to listen for. This signal is triggered when `nautobot-server migrate` or `nautobot-server post_upgrade` is run after installing an app, and provides an opportunity for the app to make any desired additions to the database at this time.
```

Comment:
Having just gone through updating a *bunch* of app signals (autocreating statuses, roles, custom fields, relationships, etc.) from Nautobot v1 to v2 for a customer, I'm no longer certain this is the best practice we want to recommend. If the data population had been implemented via data migrations instead, the existing data would have been automatically and correctly migrated to v2 data models as a part of any upgrade; because the signals run every time the app starts, they all had to be modified/rewritten to account for the v2 data model changes. Just food for thought.

Reply:
I like the general idea. There is another point about unit testing there, I've found some weird edge cases where sometimes with `--keepdb` the signal-handler-created objects just disappear and you end up troubleshooting weird issues. Do we have real world experience using data migrations over signal handlers? I want to make sure that we don't recommend this now and then wind up finding out it is equally or more annoying to deal with these.

Reply:
That's a fair question. We know that the `Status` and `Role` objects created by data migrations in core don't stick around when using `TransactionTestCase` (https://docs.djangoproject.com/en/4.2/topics/testing/overview/#rollback-emulation) (it would be interesting to experiment with `serialized_rollback = True` but I don't believe that's something we've explored...) but otherwise I'd expect them to be present - but as you say, I'm not sure how much hands-on evidence we have about that specific concern.

Reply:
Okay, since we have scrapped having a recommendation now I think I would be fine resolving this, WDYT?